stripped/colabs/GrupowanieEM_stripped.py

```diff
@@ -0,0 +1,396 @@
+NASA2020an = add_column('Albedo', 6, NASA_2020_set, NASA2020an)
+NASA2020an = add_column('Albedo', 7, NASA_2020_set, NASA2020an)
+NASA2020an = add_column('Albedo', 8, NASA_2020_set, NASA2020an)
+NASA2020an = add_column('Albedo', 9, NASA_2020_set, NASA2020an)
+
+for i in range(4, 10):
+    NASA2020an = add_column('RootMoist', i , NASA_2020_set, NASA2020an)
+
+NASA2020an.head(5)
+
+
+NASA2020features = NASA2020an.drop(columns=['lon', 'lat'])
+
+#standaryzujemy dane:
+scaler = preprocessing.StandardScaler()
+scaler.fit(NASA2020features)
+standarized = scaler.transform(NASA2020features)
+
+st_NASA2020 = pd.DataFrame(standarized, columns=NASA2020features.columns)
+st_NASA2020.head()
+
+gm = GaussianMixture(n_components = 2, n_init = 300, max_iter=200, init_params= 'k-means++', covariance_type='spherical', random_state=42)
+gm_result = gm.fit_predict(st_NASA2020)
+
+labels_ = labels['pustynia'].tolist()
+
+accuracy_score(gm_result, labels_)
+
+precision_score(gm_result, labels_)
+
+jaccard_score(gm_result, labels_)
+
+stNASA2020_ver2 = st_NASA2020.drop(columns = ['GVEG1', 'GVEG2', 'GVEG12'])
+
+gm = GaussianMixture(n_components = 2, n_init = 300, max_iter=200, init_params= 'random_from_data', covariance_type='spherical', random_state=42)
+gm_result = gm.fit_predict(stNASA2020_ver2)
+
+print(accuracy_score(gm_result, labels_))
+print(precision_score(gm_result, labels_))
+print(jaccard_score(gm_result, labels_))
+
+
+NASA_2020_set
+
+def gmm_monthly_data(month: int, df: pd.DataFrame) -> Tuple[float, float, float, List[int]]:
+
+  NASA_M_2020 = df[df['Month']==month]
+  NASA_month = NASA_M_2020.drop(columns=['lon', 'lat', 'pustynia', 'step', 'Month', 'SoilT_40_100cm', 'SoilM_100_200cm'])
+  scaler = preprocessing.StandardScaler()
+  scaler.fit(NASA_month)
+  standarized = scaler.transform(NASA_month)
+  st_NASA_month = pd.DataFrame(standarized, columns=NASA_month.columns)
+  gm = GaussianMixture(n_components = 2, n_init = 100, max_iter=100, init_params= 'random_from_data', covariance_type='spherical')
+  gm_result = gm.fit_predict(st_NASA_month)
+  label = NASA_M_2020['pustynia'].tolist()
+  acc = accuracy_score(gm_result, label)
+
+  if acc <0.5:
+    acc = 1 - acc
+    gm_result = gm_result.tolist()
+    for i in range(len(gm_result)):
+      gm_result[i] = 1 - gm_result[i]
+
+  pre = precision_score(gm_result, label)
+  jac = jaccard_score(gm_result, label)
+  return (round(acc, 3), round(pre, 3), round(jac, 3), gm_result)
+
+for i in range(12):
+  acc, pre, jac, labels = gmm_monthly_data(i+1, NASA_2020_set)
+  print("Accuracy dla danych z miesiąca ", i+1, " wynosi ", acc, " , Precision: ", pre, ", Jaccard score: ", jac )
+
+
+labels1 = np.zeros(500)
+months = [5,6,7,8,9,11]
+for i in range(6):
+  acc, pre, jac, labels2 = gmm_monthly_data(months[i], NASA_2020_set)
+  for j in range(len(labels1)):
+     labels1[j] = labels1[j] + labels2[j]
+
+for j in range(len(labels1)):
+  if labels1[j]>=4:
```

Comment:
4 to magic-number i ukryta niejawna zależność.

Reply:
Może po prostu wyrzucić wogóle tę część kodu? To był tylko eksperyment, i raczej z niego nie skorzystamy - po treningu na calym datasecie ewaluacja na podzbiorze anotowanym dała acuraccy poniżej 80% (chyba 76-77), a o precision_score już nawet nie wspomnę (nie pamiętam już, ale na pewno ponizej 0.5 ).

Reply:
up to you